apps/src/lib/client/rpc.rs

```diff
@@ -1587,9 +1553,45 @@ pub async fn get_proposal_votes(
                 .await;
                 if let Some(amount) = delegator_token_amount {
                     if vote.is_yay() {
-                        yay_delegators.insert(voter_address, amount);
+                        match yay_delegators.get_mut(&voter_address) {
+                            Some(map) => {
+                                map.insert(
+                                    validator_address,
+                                    VotePower::from(amount),
+                                );
+                            }
+                            None => {
+                                let delegations_map: HashMap<
+                                    Address,
+                                    VotePower,
+                                > = HashMap::from([(
+                                    validator_address,
+                                    VotePower::from(amount),
+                                )]);
+                                yay_delegators
+                                    .insert(voter_address, delegations_map);
+                            }
+                        }
```

Comment:
you can write these with:


```suggestion
                        let entry = yay_delegators.entry(voter_address).or_default();
                        entry.insert(validator_address, VotePower::from(amount));
```